src/test/java/codeu/controller/LoginServletTest.java

```diff
@@ -76,31 +66,54 @@ public void testDoPost_NewUser() throws IOException, ServletException {
 
     loginServlet.doPost(mockRequest, mockResponse);
 
-    ArgumentCaptor<User> userArgumentCaptor = ArgumentCaptor.forClass(User.class);
-
-    Mockito.verify(mockUserStore).addUser(userArgumentCaptor.capture());
-    Assert.assertEquals(userArgumentCaptor.getValue().getName(), "test username");
-
-    Mockito.verify(mockSession).setAttribute("user", "test username");
-    Mockito.verify(mockResponse).sendRedirect("/conversations");
+    Mockito.verify(mockUserStore, Mockito.never()).addUser(Mockito.any(User.class));
+    Mockito.verify(mockRequest).setAttribute("error", "That username was not found.");
+    Mockito.verify(mockRequestDispatcher).forward(mockRequest, mockResponse);
   }
 
   @Test
   public void testDoPost_ExistingUser() throws IOException, ServletException {
     Mockito.when(mockRequest.getParameter("username")).thenReturn("test username");
-
+    Mockito.when(mockRequest.getParameter("password")).thenReturn("test password");
+    
     UserStore mockUserStore = Mockito.mock(UserStore.class);
     Mockito.when(mockUserStore.isUserRegistered("test username")).thenReturn(true);
-    loginServlet.setUserStore(mockUserStore);
+    User mockUser = Mockito.mock(User.class);
+    Mockito.when(mockUser.getName()).thenReturn("test username");
+    Mockito.when(mockUser.getPassword()).thenReturn("test password");
+    Mockito.when(mockUserStore.getUser("test username")).thenReturn(mockUser);
 
+    loginServlet.setUserStore(mockUserStore);
+    
     HttpSession mockSession = Mockito.mock(HttpSession.class);
     Mockito.when(mockRequest.getSession()).thenReturn(mockSession);
-
+    
     loginServlet.doPost(mockRequest, mockResponse);
 
     Mockito.verify(mockUserStore, Mockito.never()).addUser(Mockito.any(User.class));
-
     Mockito.verify(mockSession).setAttribute("user", "test username");
     Mockito.verify(mockResponse).sendRedirect("/conversations");
   }
+  
+  @Test
+  public void testDoPost_WrongPassword() throws IOException, ServletException {
+	Mockito.when(mockRequest.getParameter("username")).thenReturn("test username");
+    Mockito.when(mockRequest.getParameter("password")).thenReturn("test password");
+    
+    UserStore mockUserStore = Mockito.mock(UserStore.class);
+    Mockito.when(mockUserStore.isUserRegistered("test username")).thenReturn(true);
+    
+    User mockUser = Mockito.mock(User.class);
+    Mockito.when(mockUser.getName()).thenReturn("test username");
+    Mockito.when(mockUser.getPassword()).thenReturn("wrong password");
+    Mockito.when(mockUserStore.getUser("test username")).thenReturn(mockUser);
```

Comment:
This line should be above the mock.when lines that uses mockUser.  Since Mockito.when only sets an expectation without defining order, this code runs.  This is again for test reaability.